### PR TITLE
SIT-2898: Add divnull and nullifzero

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SIT-2898

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://github.com/snowflakedb/snowpark-python/blob/main/CONTRIBUTING.md#thread-safe-development)

3. Please describe how your code solves the related issue.

   Add functions - `divnull` and `nullifzero`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@
 - Added support for `Series.str.ljust` and `Series.str.rjust`.
 - Added support for `Series.str.center`.
 - Added support for `Series.str.pad`.
+- Added support for the following functions
```
Comment (sfc-gh-yuwang) [inline CHANGELOG.md:12]: what about Added support for following functions in `functions.py`:

**Comment by sfc-gh-yuwang** [comment]
I think we also need test for it? Otherwise it won't pass combine coverage

File: src/snowflake/snowpark/functions.py [hunk 2]
```diff
@@ -2014,6 +2014,47 @@ def div0(
...
+        lit(dividend, _emit_ast=False)
+        if isinstance(dividend, (int, float))
+        else _to_col_if_str(dividend, "divnull")
+    )
+    divisor_col = (
+        lit(divisor, _emit_ast=False)
+        if isinstance(divisor, (int, float))
+        else _to_col_if_str(divisor, "divnull")
+    )
+    return dividend_col / nullifzero(divisor_col, _emit_ast=False)
```
Comment (sfc-gh-mrojas) [inline src/snowflake/snowpark/functions.py:2042]: The problem with this is that in there are valid cases of divisoin that return 0, for example 0/100, 0/20

and with this approach we wont be able to distinguish those from failed ones. Also this we can parity with spark behavior

Comment (sfc-gh-mrojas) [reply src/snowflake/snowpark/functions.py:2042]: I just understood the nullifzero on the divisor that is great as divisions by zero will then be null

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-aalam) [inline CHANGELOG.md:12]:
```suggestion
- Added support for the following functions in `functions.py`
```

**Comment by sfc-gh-aalam** [comment]
> I think we also need test for it? Otherwise it won't pass combine coverage

I'm using doc tests to fulfill the coverage. Let's see if it fails. I'll add more test cases then.

**Review by sfc-gh-yuwang — APPROVED** [review]

**Review by sfc-gh-aling — APPROVED** [review]

**Comment by sfc-gh-aherreraaguilar** [comment]
Hello @sfc-gh-aalam , I was checking this with @sfc-gh-omendezarias and realized that already exists the builtIn [div0null](https://docs.snowflake.com/en/sql-reference/functions/div0null) in snowflake? Is it better to use that instead?
Also, the name should match with the matching snowflake function "DIV0NULL"

**Pull request merged**